.github/workflows/rust.yml

```diff
@@ -54,3 +56,29 @@ jobs:
     - run: cargo hack --feature-powerset check --target ${{ matrix.sys.target }}
     - if: matrix.sys.test
       run: cargo hack --feature-powerset test --target ${{ matrix.sys.target }}
+
+  cargo-publish:
+    if: github.event_name == 'release'
+    needs: [complete]
+    runs-on: ubuntu-latest
+    steps:
+    - if: contains(needs.*.result, 'failure')
+      run: exit 1
```

Comment:
This shouldn't be necessary. Jobs will only unconditionally run if they have `if: always()`. Are you finding this is necessary for some reason?

Reply:
I thought jobs were unconditionally run (or rather the success/failure of their `needs` jobs are ignored) if they have _any_ `if:` condition

Reply:
Github docs say "If a job fails, all jobs that need it are skipped unless the jobs use a conditional expression that causes the job to continue" which suggests to me that the `if:` I have there _would_ allow the job to continue unless I inspect its `needs` otherwise.

Reply:
>You can use the following status check functions as expressions in if conditionals. A default status check of success() is applied unless you include one of these functions.

Ref: https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions

Unless the `if:` references one of the four status functions GitHub automatically adds `success()` as a condition. Unless I'm missing something, I don’t think you need the manual exit 1 failure in these jobs.

Reply:
fascinating. removed!